File: 백준/Silver/2839. 설탕 배달/설탕 배달.py
n = int(input())
max5kg = n//5
min_val = 10000
while max5kg >= 0:
    tmp = n # 초기화
    count = 0 # 초기화
    count += max5kg
    tmp -= max5kg*5
    count += tmp//3
    if tmp%3 == 0:
        min_val = min(min_val,count)
    max5kg-=1
if min_val == 10000 :
    print(-1)
else:
    print(min_val)

File: 백준/Silver/2839. 설탕 배달/설탕 배달.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int sugar = sc.nextInt();

        int minCount = Integer.MAX_VALUE;
        for (int i = 0; i < 1100; i++) {
            int count = 0;
            int val = sugar-(i*5);
            int endVal = val % 3;
            count += i;
            count += val/3;
            if(sugar-(i*5)<0) break;

            if(endVal==0 && count>0) {
                minCount = Math.min(minCount,count);
            }

        }
        if(minCount == Integer.MAX_VALUE) System.out.println(-1);
        else System.out.println(minCount);

    }

}
